docs/proposals/image-index.md

```diff
@@ -0,0 +1,150 @@
+# Image Classification API Proposal
+
+## Problem
+
+Allow to efficiently report usage of images in the cluster by tracking the
+relationship between images from the **base** image to the **application** image.
+
+By having this system in place, we can provide a better auditing of which and
+how images are used. In case of a vulnerability disclosure in one of the base
+images, we can identify the affected applications built on top of this base
+image and trigger an update.
+
+Additionally, administrator should be able to limit the usage to certain image
+classes (iow. On this cluster, you are only allowed to deploy applications
+based on *base* image).
+
+## Use Cases
+
+* Classify a certain image by a label and have that label propagated to all descentants of that
+  image currently present in the cluster.
+  * By classification, we are giving an Image context, or role in the cluster. For example an image
+    classified as "VENDOR" represents all images that are based on this "VENDOR" image.
+* Be able to query:
+  * ImageStreamTags that currently reference an Image classified as *foo*
+  * ImageStreamTags that reference outdated version of an Image classified as *foo*
+  * All Images based on the Image *foo* (descentants)
+  * All Images that the *foo* images is based on (ancestors)
+* Be able to track the usage of images limited by an license by giving the image carrying the licenced
+  content a class
+* Be able to enforce policies based on classification:
+  * Allow running only *VENDOR* based images in this cluster
```

Comment:
But the base of the image is still ubuntu (due to `FROM` clause) and that's how the image should be classified. The contents do not matter that much.